### ISSUE TITLE
Subway station name gets lost if entrance name is specified

### ISSUE THREAD

**Issue by nicolashohm** [osmandapp/OsmAnd-iOS#3402]
### Description

According to the [OSM Wiki](https://wiki.openstreetmap.org/wiki/Tag:railway%3Dsubway_entrance) the `name=*` of `railway=subway_entrance` indicates the name of the entrance not the name of the station. 

I love it that OsmAnd shows them on the map because it's very helpful to identify the correct exit to leave a subway station. 
Unfortunately, if there is an entrance name, the station name is not shown anywhere anymore on the entrance.

### Steps to reproduce

Find a subway entrance (which has a name tag) on the map in OsmAnd ([example](https://www.openstreetmap.org/node/2976098015))

### Actual result

The entrance POI is labeled "Avenue Jean Moulin" (which is the entrance name).
Swiping up the card with more details about that POI does not tell anywhere the name of the station (Torcy).

### Expected result

Show the station name in the details card for the subway_entrance, similar to other details which are shown there (ref, level, etc.).

It would probably not helpful to clutter the map with the station name, I guess in most cases it's possible to find the station POI in the map. But in some cases when it's hard to find the station POI on the map, it would be helpful to find the station name in the details.

### Your Environment (required)

```
OsmAnd Version: 4.5.5
Android/iOS version: 17.1.2
Device model: iPhone 13 Pro
```

**Comment by vshcherb** [comment]
To do: display name from main station for subway_entrance (check other exits)

**Comment by vshcherb** [comment]
To do: make same output of transport context menu as in Android, in case of doubt consult which way is more preferable.

**Comment by Dima-1** [comment]
The iOS transport context menu outputs the same as Android, except sorting routes.

**Comment by yuriiurshuliak** [comment]
The bug has been fixed. Metro station names are now displayed correctly and consistently on both iOS and Android platforms. Additionally, it appears that the context menu is now the same across both iOS and Android.

OsmAnd~ 4.8.0#2106m, released: 2024-04-17
OsmAnd 4.7.3.9

<table style="width:100%">
  </tr>
  <tr>
    <th>Android</th>
    <th>iOS</th>
    <th>Android</th>
    <th>iOS</th>
  </tr>
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/127092082/c366201b-32f2-4714-a206-056146161ff5"  width="350" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/127092082/4a43fd10-0d6a-453b-bb39-f719210bdc01"  width="400" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/127092082/fb7da8c9-5737-4e32-ba9d-5a26b2856ace"  width="350" height="auto"/> </th>
  <th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/127092082/7e6a73a4-0781-4ed6-b9d9-c7bf2130a581"  width="400" height="auto"/> </th>
  </tr>
</table>

**Comment by DmitryAlexei** [comment]
For some subway exits doesn't work. For example https://www.openstreetmap.org/node/2491871589
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/116079541/f8bf754c-bf3a-4804-8308-c6f0e5560b8e"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/116079541/b793853f-6aac-453b-b87e-557824c786fc"  width="300" height="auto"/> </td>
  </tr>
</table>

Comparing with old behavior, 
- POI doesn't display, that this is subway exit 
-  all POI, that I tested show BUS icon instead of Subway icon.
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/116079541/8df0fef3-149d-477b-9f24-312e6c04fc07"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/116079541/f8bf754c-bf3a-4804-8308-c6f0e5560b8e"  width="300" height="auto"/> </th>
 <th><img src="

**Comment by DmitryAlexei** [comment]
OsmAnd~ 4.8.0#2225m, released: 2024-05-01 and iOS 4.7.4.4
Now in Android and iOS versions Subway station name is indicated as POI name. But now
- Subway entrance name is lost
- Later in iOS it was clearly indicated that it is a **subway entrance** in POI context menu. Now in both Android and iOS versions POI is named **Stop** (icons are also from bus stop)
 
<table style="width:100%">
  <tr>
<th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/116079541/77d0167c-b8e0-4d34-8a6b-7f789eab8dfd"  width="300" height="auto"/> </th>
 <th><img src="https://github.com/osmandapp/OsmAnd-iOS/assets/116079541/bf65300a-0fc1-42b4-8c85-697dd3d58396"  width="300" height="auto"/> </td>
  </tr>
</table>
